mongodb/devhub
# DEVHUB-195: Bump Deps and Fix Lingering Peer Dependency Warnings

[Staging Link](https://docs-mongodbcom-staging.corp.mongodb.com/master/devhub/jordanstapinski/js/fix-peer-deps/)
[JIRA Ticket](https://jira.mongodb.org/browse/DEVHUB-195)

This PR updates a majority of DevHub dependencies, and removes some old peer dependency warnings called out in the linked ticket.

There are still some peer dependency warnings that show when running `npm install`. These are related to outdated dependencies in Gatsby (responsible for `@typescript-eslint/eslint-plugin@2.34.0`, `@typescript-eslint/parser@2.34.0`, `eslint-config-react-app@5.2.1`, and `gatsby-interface@0.0.166`). At this point, any dependency warnings due to dependencies we directly install have been resolved.

Breaking changes found included storybook and react-player. Storybook required a change in the config to load the theme earlier on, whereas react-player required a new callback to be specified called `onReady`. This field triggers when the player is ready to go, so we should wait for it before beginning to use the player. This only was relevant for the podcast player, since we toggle the `playing` prop.

### To Verify

- Check that articles still look okay
- Check that videos and podcasts still play off of the learn page
- Basically nothing should look out of place
